src/common/xattr.h

```diff
@@ -19,12 +19,14 @@
 extern "C" {
 #endif
 
+#if defined(__linux__)
```

Comment:
@tchaikov 
(forget my previous message)

I looked at it again, and it is foolish to put in for anything else than Linux.
If this case would arise under FreeBSD (or other  OS)  I'd rather have it signal an error because at this point we are very likely going to patch up ENOATTR with the wrong value.
So rather than doing that, I is better for the compiler to tell that ENOATTR is not defined. 
The way to fix that is to include include/compat.h.

If you prefer I can add a compiler `#WARNING` to that extend in the `#else` clause.